accepted/2020/mono-convergence/globalization-support-in-.net5.md

```diff
@@ -0,0 +1,82 @@
+* [System.Private.CoreLib/src/System/Globalization/JapaneseCalendar.Unix.cs](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Globalization/JapaneseCalendar.Unix.cs)
+
+Existing test coverage checks the implementation which is available. It does not have any expected list of data included therefore does not check whether same set of data is available everywhere.
+
+## .NET Globalization Invariant Mode
+There is already support for the globalization [invariant mode](https://github.com/dotnet/runtime/blob/master/docs/design/features/globalization-invariant-mode.md) since .NET Core 2.0. This will be the most size effective globalization mode for .NET5 which can be set by the developers. We are going to use existing logic and tweak it where necessary to allow IL Linker to remove any globalization code from the final app when running in invariant mode.
+
+## Size Sensitive Platforms
+Existing Xamarin platforms allow customers to choose which set of collation data they would like to add to their apps if any. This still makes sense for .NET5 as on some of the platform ICU dependency is not available from the system and we’ll need to bundle it to the final app. We could make this available to everyone but it’s most likely we will focus only on platforms which are size sensitive in .NET5 time frame.
+
+### Android
+Android has historically relied on ICU for some kind of globalization support.  From basically the beginning, `Locale` , `Character`, and some classes in `java.text` were provided in the platform globalization.  If you needed more than that, you could access `libicuuc` directly or choose to bundle another version in your APK.  That was pretty much the story up until API level 23.
+
+In API level 24 and higher, the platform white lists its version of `libicuuc` and also includes a subset of [ICU4J](https://developer.android.com/guide/topics/resources/internationalization#relation) API’s.  It’s important to note that [ICU4C](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/) (accessing `libicuuc` directly) is not part of the Android NDK.  What this means is that OEM’s are not required to follow any type of rules and they could limit / alter the data files as they see fit.  The potential for undefined behavior and additional testing on our part is very real. 
+
+Consult this [issue](https://github.com/android/ndk/issues/548#issuecomment-395561629) for detailed commentary on the challenges of making ICU an NDK API.
+
+### What Does This Mean For Us?
+The good news is that p-invoking from managed code works.  Assuming `libicuuc` continues to be white listed, this should not be a problem.  It would be wise if we explored using the `ICU4J`  API subset that’s exposed in the event they decide to tighten things up.  
+
+To summarize:
+
+* We can p-invoke into the system `libicuuc`.  Globalization for .net5 should work normally
```

Comment:
What is the difference between `libicu` and `libicuuc`? Do they have the same capabilities and API?

Reply:
`libicuuc` is the ICU data. `libicu` is the code.

Reply:
to clarify libicuuc is the common library for accessing the data. https://docs.oracle.com/cd/E36784_01/html/E36873/libicuuc-3lib.html

Reply:
@danmosemsft where did you see `libicu`?

Reply:
Not in this document. But usually we talk about `libicu` and here I see `libicuuc`. I am naively asking what the difference is.

Reply:
I believe it meant the same thing. we may be referring to is as libicu for simplicity. the link http://userguide.icu-project.org/packaging is useful for how ICU4C is packaged.

Reply:
ah - thank you